docs/proposals/horizontal-pod-autoscaler.md

```diff
@@ -0,0 +1,272 @@
+Scale subresource will be supported for replication controllers and deployments.
+HorizontalPodAutoscaler object will be bound with exactly one Scale subresource and will be
+autoscaling associated replication controller/deployment through it.
+
+Scale subresource will be present for replication controller or deployment under the following paths:
+
+```api/vX/replicationcontrollers/myrc/scale```
+
+```api/vX/deployments/mydeployment/scale```
+
+It will have the following structure:
+
+```go
+// Scale subresource, applicable to ReplicationControllers and (in future) Deployment.
+type Scale struct {
+	api.TypeMeta
+	api.ObjectMeta
+
+	// Spec defines the behavior of the scale.
+	Spec ScaleSpec
+
+	// Status represents the current status of the scale.
+	Status ScaleStatus
+}
+
+// ScaleSpec describes the attributes a Scale subresource
+type ScaleSpec struct {
+	// Replicas is the number of desired replicas.
+	Replicas int
+}
+
+// ScaleStatus represents the current status of a Scale subresource.
+type ScaleStatus struct {
+	// Replicas is the number of actual replicas.
+	Replicas int
+
+	// Selector is a label query over pods that should match the replicas count.
+	Selector map[string]string
+}
+
+```
+
+Writing ```ScaleSpec.Count``` will resize the replication controller/deployment associated with
+the given Scale subresource.
+```ScaleStatus.Count``` will report how many pods are currently running in the replication controller/deployment,
+and ```ScaleStatus.PodSelector``` will return selector for the pods.
+
+## HorizontalPodAutoscaler Object
+
+We will introduce HorizontalPodAutoscaler object, it will be accessible under:
+
+```
+api/vX/horizontalpodautoscalers/myautoscaler
+```
+
+It will have the following structure:
+
+```go
+// HorizontalPodAutoscaler represents the configuration of a horizontal pod autoscaler.
+type HorizontalPodAutoscaler struct {
+	api.TypeMeta
+	api.ObjectMeta
+
+	// Spec defines the behaviour of autoscaler.
+	Spec HorizontalPodAutoscalerSpec
+
+	// Status represents the current information about the autoscaler.
+	Status HorizontalPodAutoscalerStatus
+}
+
+// HorizontalPodAutoscalerSpec is the specification of a horizontal pod autoscaler.
+type HorizontalPodAutoscalerSpec struct {
+	// ScaleRef is a reference to Scale subresource. HorizontalPodAutoscaler will learn the current
+	// resource consumption from its status, and will set the desired number of pods by modyfying its spec.
+	ScaleRef *SubresourceReference
+	// MinCount is the lower limit for the number of pods that can be set by the autoscaler.
+	MinCount int
+	// MaxCount is the upper limit for the number of pods that can be set by the autoscaler.
+	// It cannot be smaller than MinCount.
+	MaxCount int
+	// Target is the target average consumption of the given resource that the autoscaler will try
+	// to maintain by adjusting the desired number of pods.
+	// Currently two types of resources are supported: "cpu" and "memory".
+	Target ResourceConsumption
```

Comment:
To make it more explicit that you only support one resource at a time, I would say: Currently this can be either "cpu" or "memory"